views.go

```diff
@@ -21,6 +24,30 @@ import (
 
 const maxLogBytes = 4 * 1024 * 1024 // 4 MB
 
+const (
+	corsOriginsEnvVariable = "LK_CORS_ORIGINS"
+)
+
+func getCorsOrigins() []string {
+	origins := os.Getenv(corsOriginsEnvVariable)
+	if origins == "" {
+		return []string{}
+	}
+	return strings.Split(origins, ",")
+}
+
+func addCorsHeaders(w http.ResponseWriter, r *http.Request) {
+	requester := r.Header.Get("Origin")
+	// check if requester is in cors origins list
+	if utility.StringMatchesAnyRegex(requester, getCorsOrigins()) {
+		w.Header().Add("Access-Control-Allow-Origin", requester)
+		w.Header().Add("Access-Control-Allow-Credentials", "true")
+	} else {
+		// Maintain backwards compatibility with the old CORS header
+		w.Header().Add("Access-Control-Allow-Origin", "*")
```

Comment:
why is the "Access-Control-Allow-Credentials" header not necessary for other hosts?

Reply:
This was the old behavior of the app i opted to not change it.